src/main/java/vivatech/util/TierHelper.java

```diff
@@ -0,0 +1,45 @@
+package vivatech.util;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import net.minecraft.block.Block.Settings;
+import net.minecraft.item.BlockItem;
+import net.minecraft.item.Item;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.Registry;
+import vivatech.Vivatech;
+import vivatech.block.AbstractTieredMachineBlock;
+
+public class TierHelper {
+
+	public static Identifier getTieredID(Identifier id, MachineTiers tier) {
+		return new Identifier(id.getNamespace(), tier.getAffix() + "_" + id.getPath());
+	}
+	
+	public static AbstractTieredMachineBlock[] getBlockArray(Map<MachineTiers, ? extends AbstractTieredMachineBlock> map) {
+		return map.values().toArray(new AbstractTieredMachineBlock[MachineTiers.values().length]);
+	}
+
+	public static void registerTieredBlockMap(Map<MachineTiers, ? extends AbstractTieredMachineBlock> map) {
+		for(MachineTiers tier : MachineTiers.values()) {
+			AbstractTieredMachineBlock block = map.get(tier);
+			Registry.register(Registry.BLOCK, block.getTieredID(), block);
+		}
+	}
+
+	public static Map<MachineTiers, Item> fillTieredBlockItemMap(Map<MachineTiers, ? extends AbstractTieredMachineBlock> blockMap) {
```

Comment:
Again, this method could pretty much go away - nothing changes when registering a tiered machine block versus any other block, all you need is something like `public static Item register(Block, Item.Settings)` and it'll cover this case plus all the others.

Reply:
I'd like to move to a single method for registering items but I'm not sure if that's something I should include in this PR.